cpuid.h error in new Mac ARM architecture

When trying to run
```
git clone https://github.com/NeuralNetworkVerification/Marabou.git
cd path/to/marabou/repo/folder
mkdir build 
cd build
cmake ..
cmake --build .
```
On the new M1 Mac with ARM processor and MacOS Monterey 12.1 I get the following error:
```
[  0%] Building CXX object CMakeFiles/MarabouHelper.dir/src/engine/DnCManager.cpp.o
In file included from /Users/marcocasadio/Projects/Marabou/src/engine/DnCManager.cpp:20:
In file included from /Users/marcocasadio/Projects/Marabou/src/common/GetCPUData.h:23:
/Library/Developer/CommandLineTools/usr/lib/clang/12.0.5/include/cpuid.h:11:2: error: this header is for x86 only
#error this header is for x86 only
```
The problem seems to be in the `GetCPUData.h` file that imports `cpuid.h`.
The error seems to be thrown in this compiler file:
`https://opensource.apple.com/source/lldb/lldb-167.2/llvm/tools/clang/lib/Headers/cpuid.h.auto.html`
Maybe there is some way that is more cross-platform compatible?

--- a/src/common/GetCPUData.h
+++ b/src/common/GetCPUData.h
@@ -19,7 +19,7 @@
 #if defined( __linux__ )
 #include <sched.h>
 #define getCPUId( cpu ) { cpu = sched_getcpu(); }
-#elif __APPLE__
+#elif __APPLE__ && (!defined(__arm64__))
 #include <cpuid.h>
 
 #define CPUId( INFO, LEAF, SUBLEAF ) __cpuid_count( LEAF, SUBLEAF, INFO[0],  \

